File: Factory/C#/Logistics/NoFactory/Boat.cs
using System.Text;

namespace NoFactory
{
    public class Boat
    {

        private readonly string _whatToDeliver;
        private readonly Coordinate[] _transportPoints;

        public Boat(string what, Coordinate[] transportPoints)
        {
            _whatToDeliver = what;
            _transportPoints = transportPoints;
        }

        public string Transport()
        {
            var builder = new StringBuilder("Transporting by boat through points\n");

            foreach (var transportPoint in _transportPoints)
            {
                builder.Append(transportPoint);
            }

            return builder.ToString();

        }

        public string Deliver()
        {
            return $"Delivering {_whatToDeliver}s by Boat";
        }
    }
}

File: Factory/Java/LogisticsNoFactory/src/LogisticsNoFactory/Boat.java
package LogisticsNoFactory;

import LogisticsNoFactory.Coordinate;

public class Boat {

	private String _whatToDeliver;
	private Coordinate[] _transportPoints;
	
	public Boat(String what, Coordinate[] transportPoints) {
		_whatToDeliver = what;
		_transportPoints = transportPoints;
	}
	
	public String transport(){
		StringBuilder builder = new StringBuilder("Transporting by boat through points\n");
		for(Coordinate c : _transportPoints)
			builder.append(c.toString());
		
		return builder.toString();
		
	}
	
	public String deliver()
	{
		return String.format("Delivering %s by Boat",_whatToDeliver);
	}
}
